homeassistant/components/modbus/entity.py

```diff
@@ -112,14 +112,16 @@ def get_optional_numeric_config(config_name: str) -> int | float | None:
     async def _async_update(self) -> None:
         """Virtual function to be overwritten."""
 
-    async def async_update(self) -> None:
+    async def async_update(self, now: datetime | None = None) -> None:
         """Update the entity state."""
-        if self._cancel_call:
-            self._cancel_call()
-        await self.async_local_update()
+        await self.async_local_update(restart=True)
 
-    async def async_local_update(self, now: datetime | None = None) -> None:
+    async def async_local_update(
+        self, now: datetime | None = None, restart: bool = False
```

Comment:
Can we make restart more descriptive? Like now that I read this function I think I understand (I am still not sure) but from reading a platform I would not know at all what that would mean

Reply:
Right, thanks for the explanation. I am wondering if we should invert the question and call it `cancel_running_update_task` or something along the lines. Like I am unable to think of a good name that describes the inverse, but this might make sense?

Reply:
I do not agree, but I have no problem changing it, I prefer compact but readable code. To me "restart" or "retrigger" tells it all, just read it with the method name "retrigger async_update", that tells me more than your long name.

Reply:
I have no problem with your opinion, but please tell me, what name you want and I will change it.

Reply:
Done, changing.

What is retriggered is the scan_interval, if you call it with false, the scan_interval update will continue uninterrupted.